otx/algorithms/detection/adapters/mmdet/task.py

```diff
@@ -0,0 +1,829 @@
+
+    def _infer_model(
+        self,
+        dataset: DatasetEntity,
+        inference_parameters: Optional[InferenceParameters] = None,
+    ):
+        """Main infer function."""
+        self._data_cfg = ConfigDict(
+            data=ConfigDict(
+                train=ConfigDict(
+                    otx_dataset=None,
+                    labels=self._labels,
+                ),
+                test=ConfigDict(
+                    otx_dataset=dataset,
+                    labels=self._labels,
+                ),
+            )
+        )
+
+        dump_features = True
+        dump_saliency_map = not inference_parameters.is_evaluation if inference_parameters else True
+
+        self._init_task()
+
+        # deepcopy all configs to make sure
+        # changes under MPA and below does not take an effect to OTX for clear distinction
+        recipe_cfg = deepcopy(self._recipe_cfg)
+        data_cfg = deepcopy(self._data_cfg)
+        assert recipe_cfg is not None, "'recipe_cfg' is not initialized."
+
+        # update model config -> model label schema
+        data_classes = [label.name for label in self._labels]
+        model_classes = [label.name for label in self._model_label_schema]
+
+        recipe_cfg.work_dir = self._output_path
+        recipe_cfg.resume = self._resume
+
+        cfg = self.configure(recipe_cfg, self._model_ckpt, data_cfg, False, "test", None, data_classes, model_classes)
+        logger.info("infer!")
+
+        samples_per_gpu = cfg.data.test_dataloader.get("samples_per_gpu", 1)
+        if samples_per_gpu > 1:
+            # Replace 'ImageToTensor' to 'DefaultFormatBundle'
+            cfg.data.test.pipeline = replace_ImageToTensor(cfg.data.test.pipeline)
+
+        # Data loader
+        dataset = build_dataset(cfg.data.test)
+        dataloader = build_dataloader(
+            dataset,
+            samples_per_gpu=cfg.data.get("samples_per_gpu", 1),
+            workers_per_gpu=cfg.data.get("workers_per_gpu", 0),
+            num_gpus=len(cfg.gpu_ids),
+            dist=cfg.distributed,
+            seed=cfg.get("seed", None),
+            shuffle=False,
+        )
+
+        # Target classes
+        if "task_adapt" in cfg:
+            target_classes = cfg.task_adapt.final
+            if len(target_classes) < 1:
+                raise KeyError(
+                    f"target_classes={target_classes} is empty check the metadata from model ckpt or recipe "
+                    "configuration"
+                )
+        else:
+            target_classes = dataset.CLASSES
+
+        # Model
+        model = build_detector(cfg)
+        if getattr(cfg, "fp16", False):
+            wrap_fp16_model(model)
+        model.CLASSES = target_classes
+        model.eval()
+        feature_model = model.model_t if self._train_type == TrainType.SEMISUPERVISED else model
+        model = build_data_parallel(model, cfg, distributed=False)
+
+        # InferenceProgressCallback (Time Monitor enable into Infer task)
```

Comment:
Could you explain why these lines are necessary? As far as I know, time_monitor is already added by OTXProgressHook. So it can be executed without these lines.

Reply:
I just bring old codes from OTX1.0, If you think this is deprecated, then we can remove this

Reply:
It could be necessary. I will check it in my side and then, I will tell you.